Update upgrading.md (#1407)
spiffe/spire: doc/upgrading.md

* SPIRE Servers are at both 0.9.3 and 0.9.2
* SPIRE Agents are supported at 0.8.0 through 0.9.2

### SPIRE Plugin Compatibility
SPIRE plugins generally follow the same overall guarantees as all other SPIRE components with small exception for changes made to external plugins outside of SPIRE's control.

#### Configuration and Behavior Compatibility

> A built-in plugin undergoing a backwards incompatible change (e.g. change to configuration semantics, change to selectors produced, etc.) should log a warning but otherwise maintain backwards compatibility for one minor version after the change is introduced, giving operators time to adopt requisite changes.

Reviewer: s/should log a warning/will log a warning/

SPIRE cannot make any guarantees around configuration or behavior compatibility for external plugins.

#### Interface Compatibility

> All efforts will be made to introduce changes to plugin interfaces in a backwards compatible manner. If a breaking change is introduced to a plugin interface, existing plugins compiled against the old interface will still continue to function for one minor version release cycle to give operators time to adopt requisite changes. SPIRE will log warnings to make operators aware of the change.

Reviewer: Suggestion to strike first sentence - I think we want to make some hard guarantees here

```suggestion
When a breaking change is introduced to a plugin interface, existing plugins compiled against the old interface will still continue to function for one minor version release cycle to give operators time to adopt requisite changes. SPIRE will log warnings to make operators aware of the change.
```

* SPIRE Servers are at both 0.9.3 and 0.9.2
* SPIRE Agents are supported at 0.8.0 through 0.9.2

### SPIRE Plugin Compatibility
SPIRE plugins follow the same overall guarantees as all other SPIRE components. A plugin compiled against SPIRE N-1 will still work, but will log a warning. Plugins compiled against SPIRE N-2 will not work.

> 

Reviewer 2: I think we could add some clarity here that this only applies to plugins that undergo a backwards incompatible change. A plugin that remains unchanged should continue to work indefinitely (assuming it continues to adhere to the plugin contract).

What about something like the following?

```suggestion
### SPIRE Plugin Compatibility
SPIRE plugins generally follow the same overall guarantees as all other SPIRE components with small exception for changes made to external plugins outside of SPIRE's control.

#### Configuration and Behavior Compatibility
A built-in plugin undergoing a backwards incompatible change (e.g. change to configuration semantics, change to selectors produced, etc.) should log a warning but otherwise maintain backwards compatibility for one minor version after the change is introduced, giving operators time to adopt requisite changes.

SPIRE cannot make any guarantees around configuration or behavior compatibility for external plugins.

#### Interface Compatibility
All efforts will be made to introduce changes to plugin interfaces in a backwards compatible manner. If a breaking change is introduced to a plugin interface, existing plugins compiled against the old interface will still continue to function for one minor version release cycle to give operators time to adopt requisite changes. SPIRE will log warnings to make operators aware of the change.
```

I also tweaked the language to follow the nomenclature used in the other compatibility related sections (e.g. removed N-1 language). I also distinguished between changes to plugin interfaces v.s. plugin configuration.

Author: Thanks, @azdagron. Looks good to me.
